filecoin/retriever.go

```diff
@@ -184,6 +197,20 @@ func (retriever *Retriever) Request(cid cid.Cid) error {
 // Possible errors: ErrAllRetrievalsFailed
 func (retriever *Retriever) retrieveFromBestCandidate(ctx context.Context, retrievalId uuid.UUID, retrievalCid cid.Cid, candidates []RetrievalCandidate) error {
 	queries := retriever.queryCandidates(ctx, retrievalId, retrievalCid, candidates)
+
+	if !retriever.config.PaidRetrievals {
+		// filter out paid retrievals
+		ii := 0
+		zero := big.Zero()
+		for _, q := range queries {
```

Comment:
this is an in place filtering? I mean, I think it works, and maybe it's more memory efficient to avoid making a new slice, but also, this logic is pretty complicated to understand. I'd just make a new slice and add the filtered ones to it.